pkg/cli/cmd/bicep/generatekubernetesmanifest/generatekubernetesmanifest.go

```diff
@@ -0,0 +1,285 @@
+		Deploy:            factory.GetDeploy(),
+		Output:            factory.GetOutput(),
+	}
+}
+
+// Validate validates the inputs of the rad bicep generate-kubernetes-manifest command.
+func (r *Runner) Validate(cmd *cobra.Command, args []string) error {
+	r.FilePath = args[0]
+
+	var err error
+	r.DeploymentScope, err = cmd.Flags().GetString("deployment-scope")
+	if err != nil {
+		return err
+	}
+
+	if r.DeploymentScope == "" {
+		r.DeploymentScope = "/planes/radius/local/resourceGroups/default"
+	}
+
+	r.AzureScope, err = cmd.Flags().GetString("azure-scope")
+	if err != nil {
+		return err
+	}
+
+	r.AWSScope, err = cmd.Flags().GetString("aws-scope")
+	if err != nil {
+		return err
+	}
+
+	r.OutFile, err = cmd.Flags().GetString("outfile")
+	if err != nil {
+		return err
+	}
+
+	parameterArgs, err := cmd.Flags().GetStringArray("parameters")
+	if err != nil {
+		return err
+	}
+
+	if r.FileSystem == nil {
+		r.FileSystem = afero.NewOsFs()
+	}
+
+	parser := bicep.ParameterParser{FileSystem: r.FileSystem}
+	r.Parameters, err = parser.Parse(parameterArgs...)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run runs the rad bicep generate-kubernetes-manifest command.
+func (r *Runner) Run(ctx context.Context) error {
+	template, err := r.Bicep.PrepareTemplate(r.FilePath)
+	if err != nil {
+		return err
+	}
+
+	// create a DeploymentTemplate yaml file
+	// with the basefilename from the bicepfile
+	if r.OutFile == "" {
+		r.OutFile = strings.TrimSuffix(filepath.Base(r.FilePath), filepath.Ext(r.FilePath)) + ".yaml"
+	}
+
+	deploymentTemplate, err := r.generateDeploymentTemplate(filepath.Base(r.FilePath), template, r.Parameters)
+	if err != nil {
+		return err
+	}
+
+	err = r.createDeploymentTemplateYAMLFile(deploymentTemplate)
+	if err != nil {
+		return err
+	}
+
+	// Print the path to the file
+	r.Output.LogInfo("DeploymentTemplate file created at %s", r.OutFile)
+
+	return nil
+}
+
+// generateDeploymentTemplate generates a DeploymentTemplate Custom Resource from the given template and parameters.
+func (r *Runner) generateDeploymentTemplate(fileName string, template map[string]any, parameters map[string]map[string]any) (map[string]any, error) {
+	marshalledTemplate, err := json.MarshalIndent(template, "", "  ")
+	if err != nil {
+		return nil, err
+	}
+
+	providerConfig := r.generateProviderConfig()
+
+	marshalledProviderConfig, err := json.MarshalIndent(providerConfig, "", "  ")
+	if err != nil {
+		return nil, err
+	}
+
+	params := make(map[string]string)
+	for k, v := range parameters {
+		params[k] = v["value"].(string)
+	}
+
+	deploymentTemplate := map[string]any{
+		"kind":       "DeploymentTemplate",
+		"apiVersion": "radapp.io/v1alpha3",
+		"metadata": map[string]any{
+			"name":      fileName,
+			"namespace": "radius-system",
+		},
+		"spec": map[string]any{
+			"template":       string(marshalledTemplate),
+			"parameters":     params,
+			"providerConfig": string(marshalledProviderConfig),
+			"rootFileName":   fileName,
```

Comment:
Something I need to dig into, what is `rootFileName` for? This feels like we're taking one piece of data and using it for a different purpose that *feels* related but may not be related.